### PR TITLE
Checks results refactor part two

### PR DESCRIPTION
**Pull request by dottorblaster**
# Description
To better orchestrate the switch between legacy logic and the new checks engine, we need to refactor the ChecksResults component making the view layer better isolated.

## How was this tested?

Uni tests added, more unit tests will come

### CODE REVIEW

File: assets/js/components/ChecksResults/HintingProvider.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
```
Comment (arbulu89) [inline assets/js/components/ChecksResults/HintingProvider.jsx:1]: Why is the file named `HintingProvider`?
As far as I see, it doesn't have any relationship with the provider. It looks more some sort of placeholder

File: assets/js/components/ChecksResults/ChecksResults.jsx [hunk 2]
```diff
@@ -251,7 +112,43 @@ const ChecksResults = () => {
...
       )}
-      {pageContent}
+      <HintingProvider
+        catalogError={catalogError}
+        clusterID={clusterID}
+        hasAlreadyChecksResults={hasAlreadyChecksResults}
+        selectedChecks={cluster?.selected_checks}
+        onCatalogRefresh={dispatchUpdateCatalog}
+      >
+        {sortHosts(hosts).map(({ host_id: hostId, reachable, msg }, idx) => (
```
Comment (arbulu89) [inline assets/js/components/ChecksResults/ChecksResults.jsx:122]: Maybe `hostID` better

**Review by arbulu89 — APPROVED** [review]
It looks great @dottorblaster 
I hope it helps to work in the integration with wanda (which will require much more changes as well).
I wouldn't obsess on adding tests to this current version, as I expect "major" changes during the new integration

File: assets/js/components/ChecksResults/HintingProvider.jsx (hunk 1 above)
Comment (arbulu89) [reply assets/js/components/ChecksResults/HintingProvider.jsx:1]: `ResultsContainer`, `ResultsPlaceholder`, `ResultsTable`...
My prefence from left to right.
I find the component as the `content` container of this view.
But I understand it is difficult to find an obvious name

Comment (dottorblaster) [reply assets/js/components/ChecksResults/HintingProvider.jsx:1]: I'll go with `ResultsContainer`, keep in mind that we have a `provider` notion in our domain and with React you can implement a pattern called Provider, where a top-layer component passes data to underlying ones. Here we adopt that logic not to pass data but to encapsulate some of the rendering logic. Anyway it has nothing to do with the cloud provider thing :joy:

**Review by nelsonkopliku — APPROVED** [review]
Looks good!
Was that `.slice()` doing all the magic? :sweat_smile:

File: assets/js/components/ChecksResults/HintingProvider.jsx (hunk 1 above)
Comment (arbulu89) [reply assets/js/components/ChecksResults/HintingProvider.jsx:1]: Ok, I totally missunderstood the term hehe. It could be `ResultsProvider` them as well

**Review by EMaksy — APPROVED** [review]
Looks great, the code is now much more readable and not stuffed!

**Review by rtorrero — APPROVED** [review]
I usually don't like sequels but I might have to make a exception with this... good job :smile:

File: assets/js/components/ChecksResults/HintingProvider.test.jsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  it('should render the suggestion box', () => {
+    renderWithRouter(
+      <HintingProvider catalogError={false} hasAlreadyChecksResults={false} />
+    );
+
+    expect(screen.getByRole('button')).toHaveTextContent('Select Checks now!');
+  });
+
+  it('should render an hello', () => {
```
[the remark this answers is not in the document]
Comment (dottorblaster) [reply assets/js/components/ChecksResults/HintingProvider.test.jsx]: Damn thank you mate

**Pull request merged**